### PR TITLE
Notifications > Comment Details: show previous/next comments

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref: #17790 

In the Comment Notifications details view, the previous and next buttons now display the previous and next comments.

Note: this only works in the Comments filter. The All and Unread filters have not been updated yet.

To test:
- Enable the `notificationCommentDetails` feature.
- Go to Notifications > Comments. 
- Select a notification.
- Tap the previous and next arrows in the nav bar.
- Verify the previous and next comments are displayed. 

## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is disabled.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jkmassel** [comment]
You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17923-c8256c3</code>. IPA is available <a href='https://appcenter-filemanagement-distrib4ede6f06e.azureedge.net/0d645d63-1199-4864-bcc6-4c267f68f8a6/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=qig0k%2FC2JlF58DN8d7p4hJGSV7ZNfRl8pFI%2Fw59jKgc%3D&se=2022-02-10T21%3A43%3A03Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by jkmassel** [comment]
You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17923-c8256c3</code>. IPA is available <a href='https://appcenter-filemanagement-distrib5ede6f06e.azureedge.net/2f9c0eca-c855-4553-a2ce-a94a65d4ef00/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=2EfNvthz1jG3qMTUN0TnBn0GvTiqmBP0H7QiCTZ8wb0%3D&se=2022-02-10T21%3A44%3A49Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

File: WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift
Comment (frosty) [inline WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift]: To avoid the duplicate `notification` when calling this method (e.g. `configureWith(notification: notification)`) I would suggest naming it `configure(with notification: Notification)`. A call then becomes `configure(with: notification)`.

**Review by frosty — CHANGES_REQUESTED** [review]
Functionally this is working great for me 👍

However, I after I saw the note about the non-weak reference to the navigation delegate I did some testing with the memory graph debugger and I saw that there's a retain cycle going on – the coordinator and detail view controller were never being released if you tapped in and out of numerous comment notifications. There were also some strong self references used in the various actions used in `CommentContentTableViewCell`. I've fixed these up on a separate branch – you can see the commits here: https://github.com/wordpress-mobile/WordPress-iOS/compare/feature/17790-arrow_functionality...feature/17790-retain-cycle-fix?expand=1. Hope this helps!

File: WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift]: What? You don't appreciate ObjC + Swift mashups? 😉 Thank you. Tis been fixed. 😄

**Comment by ScoutHarris** [comment]
> I've fixed these up on a separate branch

You're a star!! Thank you!! Clearly I didn't see the problem, so yes that most definitely helped. 

I've merged your branch, so ready for another go please.

**Review by frosty — APPROVED** [review]
Changes look good! I just remembered something else I forgot to mention previously, however I don't think it should hold this PR up – are we able to do something to stop the view jumping around when you switch comment?

https://user-images.githubusercontent.com/4780/153639579-f5a172dc-73fb-4303-943d-c4b55f89e582.mov

**Comment by ScoutHarris** [comment]
Hey @frosty .

> are we able to do something to stop the view jumping around when you switch comment?

In short, @dvdchr had some issues getting the view to resize correctly after the content was loaded, so yea it jumps a bit due to refreshing. I won't be addressing that with this project, but maybe we can look at it in the future.

Thanks!

**Pull request merged**
